### PR TITLE
Normalize route paths with a prefix that has a trailing slash

### PR DESCRIPTION
**Pull request by nwoltman**
Handles trailing slashes in the route `prefix` so that:

```js
fastify.register(function (fastify, opts, next) {
  fastify.register(function (fastify, opts, next) {
    fastify.get('/route', (req, reply) => {
      reply.send({ hello: 'world' })
    })
    next()
  }, { prefix: '/inner/' })
  next()
}, { prefix: '/v1/' })
```

registers `/v1/inner/route` instead of `/v1//inner//route`.

See the discussion in #584.

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message and code follows [Code of conduct](https://github.com/fastify/fastify/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: fastify.js [hunk 1]
```diff
@@ -338,9 +338,14 @@ function build (options) {
       return instancePrefix
     }
 
-    if (pluginPrefix[0] !== '/') {
+    if (instancePrefix.endsWith('/')) {
```
Comment (jsumners) [inline fastify.js]: http://node.green/#ES2015-built-ins-well-known-symbols-Symbol-match--String-prototype-endsWith

Unfortunately, we support down to Node v4

Comment (delvedor) [reply fastify.js]: Ahum... CI is green...
In any case it can be changed with `path.lastIndexOf('/') === (path.length -1)` :)

Comment (jsumners) [reply fastify.js]: Oh, I see. v4 just doesn't support a regex as the first parameter.

**Comment by nwoltman** [comment]
Added the test 👍

**Comment by StarpTech** [comment]
Is it documented?

**Comment by nwoltman** [comment]
I don't think there needs to be documentation. This is really just a way to do what developers are already expecting. It would be too much information to describe all the ways that Fastify handles concatenating paths with leading/trailing slashes. The Node documentation for [`path.join`](https://nodejs.org/dist/latest-v9.x/docs/api/path.html#path_path_join_paths) doesn't even describe it.

File: fastify.js [hunk 2]
```diff
@@ -417,9 +422,14 @@ function build (options) {
...
-      const path = opts.url || opts.path
       const prefix = _fastify._routePrefix
-      const url = prefix + (path === '/' && prefix.length > 0 ? '' : path)
+      var path = opts.url || opts.path
+      if (path === '/' && prefix.length > 0) {
+        path = ''
+      } else if (path[0] === '/' && prefix.endsWith('/')) {
+        path = path.slice(1)
+      }
+      const url = prefix + path
```
Comment (mcollina) [inline fastify.js]: Can you please add some comments to describe this logic?

File: fastify.js [hunk 3]
```diff
@@ -338,9 +338,14 @@ function build (options) {
...
 
-    if (pluginPrefix[0] !== '/') {
+    if (instancePrefix.endsWith('/')) {
+      if (pluginPrefix[0] === '/') {
+        pluginPrefix = pluginPrefix.slice(1)
+      }
+    } else if (pluginPrefix[0] !== '/') {
       pluginPrefix = '/' + pluginPrefix
     }
+
```
Comment (mcollina) [inline fastify.js]: Can you please add some comments to describe this logic?

**Pull request merged**